Makefile.dep

```diff
@@ -691,9 +699,24 @@ USEPKG += nanocoap
 USEMODULE += gnrc_sock_udp
 endif
 
+# all boards use gpio for leds
```

Comment:
no true, some don't do that. But adding gpio as optional feature does probably not hurt, as it does not pull any module on its own, right?

Reply:
True, I hope.
Anyhow, there are some dependencies missing (e.g., some periph_spi need periph_gpio, or maybe leds), and I'd like to get this merged and then do the explicit dependencies in a later stage.
Before this PR, gpio was always built in, too.